bus/bus.go

```diff
@@ -466,6 +470,30 @@ func (b *bus) contractsSpendingHandlerPOST(jc jape.Context) {
 	}
 }
 
+func (b *bus) hostsAllowlistHandlerGET(jc jape.Context) {
+	allowlist, err := b.hdb.HostAllowlist(jc.Request.Context())
+	if jc.Check("couldn't load allowlist", err) == nil {
+		jc.Encode(allowlist)
+	}
+}
+
+func (b *bus) hostsAllowlistHandlerPUT(jc jape.Context) {
+	ctx := jc.Request.Context()
+	var req api.UpdateAllowlistRequest
+	if jc.Decode(&req) == nil {
+		for _, entry := range req.Add {
+			if jc.Check(fmt.Sprintf("couldn't add allowlist entry '%s'", entry), b.hdb.AddHostAllowlistEntry(ctx, entry)) != nil {
+				return
+			}
+		}
+		for _, entry := range req.Remove {
+			if jc.Check(fmt.Sprintf("couldn't remove allowlist entry '%s'", entry), b.hdb.RemoveHostAllowlistEntry(ctx, entry)) != nil {
```

Comment:
Is there no way to combine all of this into a single `b.hdb.UpdateAllowlist(added, removed)` call? To maybe reduce the number of interactions with the database to a minimum?